.meta/_partials/fields/_request_options.toml.erb

```diff
@@ -77,3 +77,42 @@ recommended that you do not lower this value below the service's internal timeou
 as this could create orphaned requests, pile on retries, and result in \
 duplicate data downstream.\
 """
+
+[<%= namespace %>.request.children.decrease_ratio]
```

Comment:
I'm wondering if it's cleaner to namespace these options? We already have quite a few under the `request` namespace and it's likely we'll introduce `request.rate_limit` and `request.rety` namespaces in the future. I'm leaning towards `request.concurrency` for these. So `request.in_flight_limit` would become `request.concurrency.limit` and then you'd add `request.concurrency.decrease_ratio`, etc. What do you think?

I'm happy to merge this as-is and follow up with namespace cleanups if it's easier.

Reply:
The options were already in a field named `auto_concurrency`, so I just exposed that. Does that work for (both of) you?

Reply:
:+1: I like that better.

We could still opt to namespace it with the algorithm to make things like trying out Netflix's gradient algorithm in the future. Otherwise we'll need to deal with keeping compatibility then.

Reply:
@jszwedko we precedence for a polymorphism style of switching. For example, the [`file.fingerprinting.strategy` option](https://vector.dev/docs/reference/sources/file/#strategy). Hence the `relevant_when` option in our metadata. We could change this precedent, but I'd rather remain consistent until we do.

That said, we could introduce a `request.auto_concurrency.algorithm` option in the future that defaults to `aimd`. What do you think?

Reply:
Ah, gotcha, I see. It seems like we could maintain future compatibility with other algorithms through that approach; adding a `request.auto_concurrency.algorithm` field at that point which would default to `aimd` as you note.

:+1: I think this is fine to be merged.

Reply:
I think adding an `.algorithm` option makes a lot of sense if we have more than one mechanism in the future. We would default to whichever one had the best performance under our test cases, but give the user the choice.